daemon/cmd/daemon.go

```diff
@@ -527,6 +527,8 @@ func NewDaemon(ctx context.Context, cancel context.CancelFunc, epMgr *endpointma
 		d.k8sWatcher.NodeChain.Register(&d.k8sWatcher.K8sSvcCache)
 	}
 
+	d.k8sWatcher.NodeChain.Register(d.k8sWatcher)
```

Comment:
I'm now wondering if it would be better to wrap `K8sWatcher` in a new type rather then aliasing it to make it more explicit what's going on here :thinking:

Reply:
Have you thought on simply implementing the methods of Node interface directly on the `K8sWatcher`? :thinking: